### PR TITLE
Unit tests for ManageController

### PR DESCRIPTION
**Pull request by Chris-Wooder**
Created small PR for 2 initially to get feedback before creating PR's for more.

ManageController.IndexGetAddsCorrectMessageToViewDataWhenMessageEqualsChangePasswordSuccess
ManageController.IndexGetAddsCorrectMessageToViewDataWhenMessageIdEqualsSetPasswordSuccess

### CODE REVIEW

**Comment by mgmccarthy** [comment]
@Chris-Wooder, I'm going to try to get my eyes on this today or tomorrow. Thanks.

**Comment by Chris-Wooder** [comment]
Great, i am ready to create another pull request but will wait for first review to apply any comments.  As more tests are populated I will refactor some of the repeated code out into some helper methods to centralise creating the controller for common scenarios.

**Comment by mgmccarthy** [comment]
@Chris-Wooder, first and foremost, thanks for your contribution. I've quickly looked at the two test implementations you've added and I've included some fixes in my code review for you to implement.

If you have any questions, please let me know.

**Comment by mgmccarthy** [comment]
@Chris-Wooder, please do NOT do this:

> As more tests are populated I will refactor some of the repeated code out into some helper methods to centralise creating the controller for common scenarios

[shared setup in unit test methods are not a good thing](http://www.michaelgmccarthy.com/2016/08/03/unit-testing-sut-setup-methods-and-code-maintainability/). I realize some of our codebase does have shared setup, and those unit test are very hard to unravel unless you're the initial author of the unit test.

Don't worry about repeated code for now, just worry about unit test implementation/correctness.

**Comment by Chris-Wooder** [comment]
@mgmccarthy thanks for the feedback, yes makes sense, should've paid more attention to stub names.  Implemented methods for the first 4 GET index stubs.

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs [hunk 1]
```diff
@@ -168,23 +168,61 @@ public class ManageControllerTests
...
+            var userManagerMock = MockHelper.CreateUserManagerMock();
+            var signInManagerMock = MockHelper.CreateSignInManagerMock(userManagerMock);
+
+            var mediator = new Mock<IMediator>();
+            mediator.Setup(m => m.SendAsync(It.IsAny<UserByUserIdQuery>())).ReturnsAsync(new ApplicationUser());
+
+            var controller = new ManageController(userManagerMock.Object, signInManagerMock.Object, mediator.Object);
+            controller.SetFakeUser("userId");
+
+            var applicationUserVM = new Mock<ApplicationUser>();
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs]: applicationUserVM is not used and can be removed

**Comment by mgmccarthy** [comment]
@Chris-Wooder, do you want to work on any other tests? If so, would you like to keep this PR open as a WIP and keep adding to it, or do you want to do small PR's and have them merged in as we go?

**Comment by Chris-Wooder** [comment]
@mgmccarthy, yes I am working on more, unless you think there are any other particularly high priority tests I will continue going through the stubs for the ManageController.

As for Pull Request, I don't mind, is there a preference on the project?

**Comment by mgmccarthy** [comment]
@Chris-Wooder however you want to work. I do know this controller is in desperate need of unit tests, and it's one of the harder controllers to test.

For now, let's leave this PR open as  WIP, and you can keep adding to it as much or as little as you want. 4-6 tests at a time (or at least logically grouped tests by action method name/type (POST/GET)) will help keep the code review process working nicely.

When we get to a certain threshold of code, we'll probably want to merge the PR.

Thanks

**Comment by Chris-Wooder** [comment]
Ok will carry on with ManageController

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs [hunk 2]
```diff
@@ -223,53 +238,143 @@ public void IndexGetHasHttpGetAttribute()
...
+            var mediator = new Mock<IMediator>();
+            var user = new ApplicationUser { TimeZoneId = "timeZoneId" };
+            mediator.Setup(m => m.SendAsync(It.IsAny<UserByUserIdQuery>())).ReturnsAsync(user);
+
+            var controller = new ManageController(userManagerMock.Object, signInManagerMock.Object, mediator.Object);
+            controller.SetFakeUser("userId");
+            var vM = new IndexViewModel { TimeZoneId = "differentTimeZoneId" };
+
+            //Act
+            var result = await controller.Index(vM);
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs]: you can remove `var result` here

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs [hunk 3]
```diff
@@ -223,53 +238,143 @@ public void IndexGetHasHttpGetAttribute()
...
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs]: you can remove `var result` here as well

**Comment by tonysurma** [comment]
thanks again @Chris-Wooder and @mgmccarthy let me know when good to go/merge

**Comment by Chris-Wooder** [comment]
This pull request contains all tests for the Index GET & POST Actions.  Maybe makes sense to create a new PR for next set of commits...

**Comment by tonysurma** [comment]
I would agree, moving to another PR allows this to get pulled in independent of the next batch.  If you're good to do that edit the name to remove wip and we'll merge. Thanks

**Comment by Chris-Wooder** [comment]
@tonysurma done.

**Pull request merged**